lib/saint.js

```diff
@@ -79,15 +79,23 @@ p.getJobSegment = function(fileId, pageNum, callback){
 	});
 }
 
-p.importJob = function(jobDescription, jobData, callback){
-	var self = this,
-			populateData =[];
-	while(jobData.length){
-		populateData.push(jobData.splice(0,10000));
+p.importJob = function(jobDescription, jobData, callback, jobId){
+	var self = this;
+	var populateData = spliceJobData(jobData);
+	function spliceJobData(jobData) {
+		var splicedJobData = [];
+		while(jobData.length){
+			splicedJobData.push(jobData.splice(0,10000));
+		}
+		return splicedJobData;
 	}
-
 	self.logger("info", "Creating Job");
-	this.importCreateJob(jobDescription, function(err, data){
+	if (typeof jobId === 'undefined') {
+		this.importCreateJob(jobDescription, fnCallImportPopulateJob);
+	} else {
+		fnCallImportPopulateJob(null, jobId);
+	}
+	function fnCallImportPopulateJob(err, data){
```

Comment:
The populateData.forEach has an anonymous function. You probably didn't change it to reduce the number of changed lines but in this case, you're altering and testing this function significantly and it would be wise to update it to be more readable while we're here. 

Can you remove the anonymous function?

Reply:
Sure